### PR TITLE
[9.99.x-prod] add kogito-addons-quarkus-jobs-knative-eventing and kogito-addons-persistence-jdbc extensions to builder and devmode images

### PR DESCRIPTION
**Pull request by rgdoliveira**
This is required by Parodos because we are doing an internal release and Kogito artifacts will not be available on MRRC

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- <b>(Re)run Jenkins tests</b>  
  Please add comment: <b>Jenkins [test|retest] this</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Review by jakubschwan — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Comment by rgolangh** [comment]
> @masayag @rgolangh can you please take a look? The intention is to have the libraries pre-cached for you based on this: https://github.com/parodos-dev/serverless-workflows/blob/main/.github/workflows/main.yml#L44

I think it is missing quarkus-jdbc-postgresql?

**Comment by ricardozanini** [comment]
> > @masayag @rgolangh can you please take a look? The intention is to have the libraries pre-cached for you based on this: https://github.com/parodos-dev/serverless-workflows/blob/main/.github/workflows/main.yml#L44
> 
> I think it is missing quarkus-jdbc-postgresql?

this one is available in Maven Central, so it should be downloaded automatically. Also, our persistence libs depends on it.

**Comment by masayag** [comment]
Thank you @ricardozanini, I'm ready to test it once it is available.

File: productized/logic/build-quarkus-app.sh [hunk 1]
```diff
@@ -44,11 +44,11 @@ properties_with_versions=("compiler-plugin.version:3.11.0" "surefire-plugin.vers
...
 quarkus_extensions_extra_deps="com.aayushatharva.brotli4j:native-linux-aarch64:1.12.0"
 # common extensions used by the kogito-swf-builder and kogito-swf-devmode
-quarkus_extensions="quarkus-kubernetes,smallrye-health,org.kie.kogito:kogito-quarkus-serverless-workflow:${kogito_version},org.kie.kogito:kogito-addons-quarkus-knative-eventing:${kogito_version},org.kie.kogito:kogito-addons-quarkus-microprofile-config-service-catalog:${kogito_version},org.kie.kogito:kogito-addons-quarkus-kubernetes:${kogito_version},org.kie.kogito:kogito-addons-quarkus-knative-serving:${kogito_version}"
+quarkus_extensions="quarkus-kubernetes,smallrye-health,org.kie.kogito:kogito-quarkus-serverless-workflow:${kogito_version},org.kie.kogito:kogito-addons-quarkus-knative-eventing:${kogito_version},org.kie.kogito:kogito-addons-quarkus-microprofile-config-service-catalog:${kogito_version},org.kie.kogito:kogito-addons-quarkus-kubernetes:${kogito_version},org.kie.kogito:kogito-addons-quarkus-knative-serving:${kogito_version},org.kie.kogito:kogito-addons-quarkus-jobs-knative-eventing:${kogito_version},org.kie.kogito:kogito-addons-persistence-jdbc:${kogito_version}"
```
Comment (wmedvede) [inline productized/logic/build-quarkus-app.sh:47]: By adding the kogito-addons-persistence-jdbc to the swf-builder directly, you are forcing that all workflows produced by the swf-builder must have persistence. And if not, the produced build will never start, which means that it's not possible to have workflows without persistence. Also, If I'm not wrong, unlike adding an extension at builder, when an extension was added into the builder in this way, it's not possible to produce a build by not using that extension. So, basically we are saying that every build produced by the builder must have persistence.
Is this behavior what we want?

@rgdoliveira @ricardozanini  @jakubschwan ?

**Pull request merged**